include/insns/RV32I.h

```diff
@@ -251,75 +251,85 @@ namespace SST{
 
       // Standard instructions
       static bool lui(RevFeature *F, RevRegFile *R,RevMem *M,RevInst Inst) {
+        uint32_t Tmp = Inst.imm << 12;
         if( F->IsRV32() ){
-          R->RV32[Inst.rd] = 0x00;
-          R->RV32[Inst.rd] = (Inst.imm << 12);
+          R->RV32[Inst.rd] = Tmp;
           R->RV32_PC += Inst.instSize;
         }else{
-          R->RV64[Inst.rd] = 0x00;
-          R->RV64[Inst.rd] = (Inst.imm << 12);
+          SEXT(R->RV64[Inst.rd], Tmp, 32);
```

Comment:
this is incorrect.  the spec says that LUI loads unsigned immediates into the top 20 bits of the target.  
```
LUI (load upper immediate) is used to build 32-bit constants and uses the U-type format. LUI places the U-immediate value in the top 20 bits of the destination register rd, filling in the lowest 12 bits with zeros.
```